controller/program/src/instructions/admin/add_lst.rs

```diff
@@ -0,0 +1,161 @@
+        .with_admin(&pool.admin)
+        .with_payer(abr.get(*accs.payer()).key())
+        .with_lst_mint(lst_mint_acc.key())
+        .with_pool_reserves(&expected_pool_reserves)
+        .with_protocol_fee_accumulator(&expected_protocol_fee_accumulator)
+        .with_protocol_fee_accumulator_auth(&PROTOCOL_FEE_ID)
+        .with_sol_value_calculator(abr.get(*accs.sol_value_calculator()).key())
+        .with_pool_state(&POOL_STATE_ID)
+        .with_lst_state_list(&LST_STATE_LIST_ID)
+        .with_associated_token_program(&ATOKEN_ID)
+        .with_system_program(&SYS_PROG_ID)
+        .with_lst_token_program(token_prog)
+        .build();
+
+    verify_pks(abr, &accs.0, &expected_pks.0)?;
+    verify_signers(abr, &accs.0, &ADD_LST_IX_IS_SIGNER.0)
+        .map_err(|expected_signer| log_and_return_acc_privilege_err(abr, *expected_signer))?;
+
+    verify_tokenkeg_or_22_mint(lst_mint_acc)?;
+    verify_sol_value_calculator_is_program(abr.get(*accs.sol_value_calculator()))?;
+
+    // Verify no duplicate in lst state list
+    let list = LstStatePackedList::of_acc_data(abr.get(*accs.lst_state_list()).data())
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidLstStateListData))?;
+
+    if list.find_by_mint(lst_mint_acc.key()).is_some() {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::DuplicateLst).into());
+    }
+
+    verify_not_rebalancing_and_not_disabled(pool)?;
+
+    // Create pool reserves and protocol fee accumulator ATAs if they do not exist
+    if abr.get(*accs.pool_reserves()).data().is_empty() {
+        let create_accs = NewCreateIxAccsBuilder::start()
+            .with_funding(*accs.payer())
+            .with_ata(*accs.pool_reserves())
+            .with_wallet(*accs.pool_state())
+            .with_mint(*accs.lst_mint())
+            .with_sys_prog(*accs.system_program())
+            .with_token_prog(*accs.lst_token_program())
+            .build();
+
+        cpi.invoke_fwd(abr, &ATOKEN_ID, CreateIxData::as_buf(), create_accs.0)?;
+    }
+
+    if abr.get(*accs.protocol_fee_accumulator()).data().is_empty() {
+        let create_accs = NewCreateIxAccsBuilder::start()
+            .with_funding(*accs.payer())
+            .with_ata(*accs.protocol_fee_accumulator())
+            .with_wallet(*accs.protocol_fee_accumulator_auth())
+            .with_mint(*accs.lst_mint())
+            .with_sys_prog(*accs.system_program())
+            .with_token_prog(*accs.lst_token_program())
+            .build();
+
+        cpi.invoke_fwd(abr, &ATOKEN_ID, CreateIxData::as_buf(), create_accs.0)?;
+    }
+
+    // Realloc lst state list
+    let lst_state_list_acc = abr.get_mut(*accs.lst_state_list());
+    let old_acc_len = lst_state_list_acc.data_len();
+    lst_state_list_acc.grow_by(size_of::<LstStatePacked>(), false)?;
+
+    let new_acc_len = lst_state_list_acc.data_len();
+
+    let lamports_shortfall = Rent::get()?
+        .min_balance(new_acc_len)
+        .saturating_sub(Rent::get()?.min_balance(old_acc_len));
```

Comment:
[`Rent::get()` is a syscall](https://github.com/igneous-labs/jiminy/blob/92fc4e4236a47e68d5ed34a1b1b2a15220424c5f/sysvar/sysvar/src/lib.rs#L34-L97) so it probably cant be optimized away by the compiler, so try to reduce calls to it.

```suggestion
    let rent = Rent::get()?;
    let lamports_shortfall = rent
        .min_balance(new_acc_len)
        .saturating_sub(rent.min_balance(old_acc_len));
```